### PR TITLE
Show upstream error body

### PR DESCRIPTION
**Pull request by tobiasbexelius**
# Jira Issue: [MX-394]

## What?
- Shows the body of the upstream call that failed
- Before:
```
{
    "errors": [
      {
        "message": "HTTP error - statusCode 400, url: http://localhost:1234/underwriter/public/onboarding-sessions",
        "locations": [
          {
            "line": 2,
            "column": 3
          }
        ],
        "path": [
          "onboardingQuoteCart_create"
        ],
        "extensions": {
          "code": "INTERNAL_SERVER_ERROR",
          "exception": {
            "errors": [
              {
                "message": "HTTP error - statusCode 400, url: http://localhost:1234/underwriter/public/onboarding-sessions",
                "locations": [],
                "path": [
                  "onboardingQuoteCart_create"
                ]
              }
            ],
            "stacktrace": [
              "Error: HTTP error - statusCode 400, url: http://localhost:1234/underwriter/public/onboarding-sessions",
              "    at new CombinedError (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/src/stitching/errors.ts:90:5)",
              "    at Object.checkResultAndHandleErrors (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/src/stitching/errors.ts:111:11)",
              "    at CheckResultAndHandleErrors.transformResult (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/src/transforms/CheckResultAndHandleErrors.ts:15:12)",
              "    at /Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/src/transforms/transforms.ts:37:45",
              "    at Array.reduce (<anonymous>)",
              "    at applyResultTransforms (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/src/transforms/transforms.ts:35:21)",
              "    at /Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/src/stitching/delegateToSchema.ts:104:12",
              "    at step (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:31:23)",
              "    at Object.next (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:12:53)",
              "    at fulfilled (/Users/tobiasbexelius/hedvig/giraffe/node_modules/graphql-tools/dist/stitching/delegateToSchema.js:3:58)"
            ]
          }
        }
      }
    ],
    "data": null
  }
```
- After:
```
{
  "errors": [
    {
      "message": "400 Bad Request",
      "locations": [],
      "path": [
        "onboardingQuoteCart_create"
      ],
      "extensions": {
        "url": "http://localhost:1234/underwriter/public/onboarding-sessions",
        "body": {
          "errorMessage": "Field 'locale2' must not be null",
          "errorCode": null
        },
        "code": "400",
        "exception": {
          "url": "http://localhost:1234/underwriter/public/onboarding-sessions",
          "body": {
            "errorMessage": "Field 'locale2' must not be null",
            "errorCode": null
          },
          "stacktrace": [
            "HttpError: 400 Bad Request",
            "    at /Users/tobiasbexelius/hedvig/giraffe/src/api/httpClient.ts:75:15",
            "    at Generator.next (<anonymous>)",
            "    at fulfilled (/Users/tobiasbexelius/hedvig/giraffe/src/api/httpClient.ts:5:58)",
            "    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
          ]
        }
      }
    }
  ],
  "data": null
}
```

## Why?
- To help developers debug

## Optional checklist
- [ ] Unit tests written
- [ ] Tested locally
- [ ] Codescouted

[MX-394]: https://hedvig.atlassian.net/browse/MX-394?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/resolvers/claims/__snapshots__/claims.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (tobiasbexelius) [inline src/resolvers/claims/__snapshots__/claims.test.ts.snap:1]: I have no idea why this didn't already exist 🤔

File: src/api/httpClient.ts [hunk 2]
```diff
@@ -13,10 +14,14 @@ export interface HttpClient {
...
-  constructor(public statusCode: number, public url: string, public body: any) {
-    super(`HTTP error - statusCode ${statusCode}, url: ${url}`)
-    this.name = 'HttpError'
+export class HttpError extends ApolloError {
+  constructor(statusCode: number, url: string, body: any) {
+    // @ts-ignore - TS is confused by this for some reason
+    super(`${statusCode} ${httpStatusMap[statusCode]}`, String(statusCode), {
+      url,
+      body
+    })
```
Comment (FredrikAreschoug) [inline src/api/httpClient.ts:23]: I don't understand why it is confused but It would be nice without the ignore

**Review by FredrikAreschoug — APPROVED** [review]
Nice improvement

File: src/api/httpClient.ts (hunk 2 above)
Comment (robinandeer) [reply src/api/httpClient.ts:23]: maybe we can clarify the comment like: "TS not able to infer that variable matches keys in map"

File: src/api/httpClient.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import fetch, { RequestInit, Response } from 'node-fetch'
 import { TokenProvider } from '../context'
+import {ApolloError} from "apollo-server-core";
```
Comment (robinandeer) [inline src/api/httpClient.ts:3]: Do you have Prettier setup? It supposed to automatically take care of consistent formatting like using single quotes. But that's only a small concern I guess.

**Review by robinandeer — APPROVED** [review]
Very nice!

File: src/api/httpClient.ts (hunk 3 above)
Comment (tobiasbexelius) [reply src/api/httpClient.ts:3]: Fixed!

**Pull request merged**